pkg/operator/controller/ingressclass/controller.go

```diff
@@ -0,0 +1,130 @@
+package ingressclass
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+
+	"k8s.io/client-go/tools/record"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	networkingv1 "k8s.io/api/networking/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName = "ingressclass_controller"
+)
+
+var (
+	log = logf.Logger.WithName(controllerName)
+)
+
+// New creates and returns a controller that creates and manages IngressClass
+// objects for IngressControllers.
+func New(mgr manager.Manager, config Config) (controller.Controller, error) {
+	reconciler := &reconciler{
+		config:   config,
+		client:   mgr.GetClient(),
+		cache:    mgr.GetCache(),
+		recorder: mgr.GetEventRecorderFor(controllerName),
+	}
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: reconciler})
+	if err != nil {
+		return nil, err
+	}
+	if err := c.Watch(&source.Kind{Type: &operatorv1.IngressController{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return nil, err
+	}
+	if err := c.Watch(&source.Kind{Type: &networkingv1.IngressClass{}}, handler.EnqueueRequestsFromMapFunc(reconciler.ingressClassToIngressControllers)); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// ingressClassToIngressControllers takes a client object, checks if it is an
+// ingressclass with an associated ingresscontroller, and returns a slice of
+// reconcile.Request with a request to reconcile any ingresscontroller that is
+// associated with the ingressclass.
+func (r *reconciler) ingressClassToIngressControllers(o client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+	classes := &networkingv1.IngressClassList{}
+	if err := r.cache.List(context.Background(), classes); err != nil {
+		log.Error(err, "failed to list ingressclasses")
+		return requests
+	}
+	for _, class := range classes.Items {
+		if !ingressClassReferencesSomeIngressController(&class) {
+			continue
+		}
+		ingresscontrollerName := class.Spec.Parameters.Name
+		if class.Spec.Parameters.Name != ingresscontrollerName {
```

Comment:
I think you meant to check the parameter ingressclass's `Spec.Parameters.Name`.

(This will always evaluate to false as is).

Reply:
Whoops!